src/core/datagram.h

```diff
@@ -103,3 +103,9 @@ QuicDatagramProcessFrame(
         const uint8_t * const Buffer,
     _Inout_ uint16_t* Offset
     );
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicDatagramCancelPending(
```

Comment:
Now that I think about it, I think we need a better name. I want this function to only cancel the datagram sends that want to be cancelled (i.e., they have the flag).